sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVRelation.scala

```diff
@@ -42,12 +42,12 @@ object CSVRelation extends Logging {
       firstLine: String,
       params: CSVOptions): RDD[Array[String]] = {
     // If header is set, make sure firstLine is materialized before sending to executors.
-    file.mapPartitionsWithIndex({
-      case (split, iter) => new BulkCsvReader(
+    file.mapPartitions { iter =>
```

Comment:
This is legitimately a little optimization, but the rest feels too minor to bother with. Really there are more instances of slightly mis-spaced scaladoc or lines of code, though to fix them all would be time-consuming. Fixing them when nearby code changes is sensible. Fixing a few at random doesn't feel like a great goal.